src/webots/nodes/WbGeometry.cpp

```diff
@@ -44,7 +44,7 @@
 // Constant used to scale down the line scale property
 const float WbGeometry::LINE_SCALE_FACTOR = 250.0f;
 
-const int gMaxIndexNumberToCastShadows = (2 << 14) - 1;  // 2^15 - 1 (16-bit resolution)
+const int gMaxIndexNumberToCastShadows = (2 << 15) - 1;  // 2^16 - 1 (16-bit resolution)
```

Comment:
Why did you increase this value?

Reply:
The shadow is disappearing when there is more than 21845 triangles, which means maximum 65535 = 2^16-1 indices. So I deduced from this there was an error in the value of `gMaxIndexNumberToCastShadows` (which was not used anyway in the original version of the code).